src/ngx_http_lua_kong_request.c

```diff
@@ -0,0 +1,109 @@
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
+
+/* name should be lowercased and preprocessed for both search functions */
+
+/* by hash */
+static ngx_str_t *search_known_header(ngx_http_request_t *r, ngx_str_t name)
```

Comment:
Also nginx style, functon's return type `static ngx_str_t*` should be at a single line.

Reply:
The function's name should have a prefix `ngx_http_`.

Reply:
> 

Do we have to add a prefix to local functions? (they are static and in file scope)

Reply:
Yes, we can refer nginx source.